provider/k8s/process.go

```diff
@@ -573,6 +575,34 @@ func (p *Provider) podSpecFromRunOptions(app, service string, opts structs.Proce
 		s.Containers[0].Resources.Limits["memory"] = resource.MustParse(fmt.Sprintf("%dMi", *opts.MemoryLimit))
 	}
 
+	p.DockerUsername = common.CoalesceString(os.Getenv("DOCKER_HUB_USERNAME"),"")
+	p.DockerPassword = common.CoalesceString(os.Getenv("DOCKER_HUB_PASSWORD"),"")
+
+	_, err = p.CreateOrPatchSecret(p.ctx, am.ObjectMeta{
```

Comment:
add a check if docker username and password is not provided, don't execute these logic